### PR TITLE
[17.0][ADD] sale_multi_template_application: new module to allow applying multiple templates to a sale order

### PR DESCRIPTION
**Pull request by carlos-lopez-tecnativa**
This module enables applying multiple quotation templates to the same sale order. When a new template is selected, the order lines from the previous templates are retained and not deleted.

TT52186
@pedrobaeza could you please review this

### CODE REVIEW

File: sale_multi_template_application/readme/DESCRIPTION.md [hunk 1]
```diff
@@ -0,0 +1 @@
+This module allows multiple templates to be applied to the same sale order, keeping the existing order lines without deleting the previous ones.
```
Comment (pedrobaeza) [inline sale_multi_template_application/readme/DESCRIPTION.md]:
```suggestion
This module allows multiple templates to be applied to the same sales order, keeping the existing order lines without deleting the previous ones.

For consistency, the template field is emptied for not keeping the order linked to a possible partial template.
```

File: sale_multi_template_application/readme/USAGE.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+- Go to *Sales \> Orders \> Quotations*.
+- Create a new sales order
```
Comment (pedrobaeza) [inline sale_multi_template_application/readme/USAGE.md]:
```suggestion
- Create a new sales order.
```

File: sale_multi_template_application/models/sale_order.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+from odoo import api, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    @api.onchange("sale_order_template_id")
+    def _onchange_sale_order_template_id(self):
+        # We need to keep the old lines to restore after the template change
```
Comment (pedrobaeza) [inline sale_multi_template_application/models/sale_order.py]:
```suggestion
        # We need to keep the old lines, to restore them after the template change
```

File: sale_multi_template_application/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
+# Copyright 2024 Tecnativa - Carlos López
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+{
+    "name": "Sale multi template application",
+    "version": "17.0.1.0.0",
+    "summary": """Allows multiple templates to be applied to the same sale order,
+    keeping the existing order lines without deleting the previous ones.""",
```
Comment (pedrobaeza) [inline sale_multi_template_application/__manifest__.py]: The summary shouldn't be so long, and most of the times, the name is enough (but a significant name):

```suggestion
    "name": "Apply multiple quotation templates",
    "version": "17.0.1.0.0",
```

**Review by pedrobaeza — APPROVED** [review]

**Review by sergio-teruel — APPROVED** [review]

**Pull request merged**